### ISSUE TITLE
[BUG] `avoid-unused-parameters` no warning when defining variable with the same name as unused parameter

### ISSUE THREAD

**Issue by DetachHead** [dart-code-checker/dart-code-metrics#977]
* **Dart code metrics version:** 4.16.0
* **Dart sdk version:** 2.17.5

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
dart_code_metrics:
  rules:
    - avoid-unused-parameters
```

</details>

**What did you do? Please include the source code example causing the issue.**

```dart
void foo(String value) {
  final value = 1;
  print(value);
}
```

**What did you expect to happen?**
`avoid-unused-parameters` warning

**What actually happened?**
no warning

**Are you willing to submit a pull request to fix this bug?**
idk

**Comment by incendial** [comment]
@DetachHead I kinda think that shadowing is not supported by this rule by design, but I'll check

**Comment by incendial** [comment]
Took a closer look and I think it's possible to fix the issue.

The rule was written 17 month ago and used the string names as a comparison target. I've rewritten the rule to use element model instead and everything looks fine.
